.github/workflows/setup_env.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+#-------------------------------------------------------------------------------
+# Functions
+
+# Suppress echo (-x) output of commands executed with `quiet`.
+# Useful for sourcing files, loading modules, spack, etc.
+# set +x, set -x are not echo'd.
+quiet() {
+    { set +x; } 2> /dev/null;
+    $@;
+    set -x
+}
+
+# `print` is like `echo`, but suppresses output of the command itself.
+# https://superuser.com/a/1141026
+echo_and_restore() {
+    builtin echo "$*"
+    date
+    case "${save_flags}" in
+        (*x*)  set -x
+    esac
+}
+alias print='{ save_flags="$-"; set +x; } 2> /dev/null; echo_and_restore'
+
+
+#-------------------------------------------------------------------------------
+quiet source /etc/profile
+
+hostname && pwd
+export top=$(pwd)
+
+shopt -s expand_aliases
+
+
+print "======================================== Load compiler"
+quiet module avail
+
+quiet module load gcc@7.3.0
+quiet which g++
+g++ --version
+
+quiet module load intel-oneapi-mkl
+echo "MKLROOT=${MKLROOT}"
+
+#quiet module load python  # no module
+quiet which python
+quiet which python3
+python  --version
+python3 --version
+
+quiet module load pkgconf
+quiet which pkg-config
+
+# CMake will find CUDA in /usr/local/cuda, so need to explicitly set
+# gpu_backend.
+export gpu_backend=none
+export color=no
+
+# For simplicity, create make.inc regardless of ${maker}
+cat > make.inc << END
+CXX    = mpicxx
+FC     = mpif90
+blas   = mkl
+prefix = ${top}/install
+md5sum = md5sum
+END
+
+if [ "${device}" = "cpu" -o "${device}" = "gpu_nvidia" ]; then
+    print "======================================== Load Open MPI"
+    module load openmpi
+    export OMPI_CXX=${CXX}
+
+    cat >> make.inc << END
+CXXFLAGS  = -Werror -Dslate_omp_default_none='default(none)'
+mkl_blacs = openmpi
+END
+else
+    print "======================================== Load Intel MPI"
+    module load intel-mpi
+    export FI_PROVIDER=tcp
+
+    # AMD has header warnings, so don't use -Werror.
+    cat >> make.inc << END
+mkl_blacs = intelmpi
+END
+fi
+print "======================================== Verify MPI"
+quiet which mpicxx
+quiet which mpif90
+mpicxx --version
+mpif90 --version
+
+if [ "${device}" = "gpu_nvidia" ]; then
+    print "======================================== Load CUDA"
+    export CUDA_HOME=/usr/local/cuda/
```

Comment:
Does it have to be hard coded?

Reply:
Turns out the container does have `module load cuda` that sets $CUDA_HOME. Sadly, there's no `module load rocm`.